android/src/main/java/com/expensify/livemarkdown/MarkdownUtils.java

```diff
@@ -66,6 +68,62 @@ public void setMarkdownStyle(@NonNull MarkdownStyle markdownStyle) {
     mMarkdownStyle = markdownStyle;
   }
 
+  private List<MarkdownRange> splitRangesOnEmojis(List<MarkdownRange> markdownRanges) {
+    List<MarkdownRange> emojiRanges = new ArrayList<>();
+    for (MarkdownRange range : markdownRanges) {
+      if (range.type.equals("emoji")) {
+        emojiRanges.add(range);
+      }
+    }
+
+    for (int i = 0; i < markdownRanges.size(); i++) {
+      MarkdownRange range = markdownRanges.get(i);
+      if (!range.type.equals("italic") && !range.type.equals("strikethrough")) {
+        continue;
+      }
+
+      for (int j = 0; j <  emojiRanges.size(); j++) {
```

Comment:
This algorithm has $O(m*n)$ time complexity where $m$ denotes the number of ranges and $n$ denotes the number of emojis. In worst case scenario, if $m=n$ (i.e. all ranges are emoji ranges), the time complexity is $O(n^2)$. Is there a better algorithm?

Reply:
Agreed, improved the algorithm time complexity to O(n) in [`8cb08b3` (#534)](https://github.com/Expensify/react-native-live-markdown/pull/534/commits/8cb08b361a81143c1c87aa9c2cf42edacb4afb05)